### PR TITLE
test: add Unit Tests to core package

### PR DESCRIPTION
**Pull request by DecarteAdam**
feat: improve core package code coverage to 90%

Added more unit tests to improve 'core' package code coverage.
Mocked ApacheHttpClient with Mockito

### CODE REVIEW

File: src/main/java/com/crowdin/client/core/http/impl/http/ApacheHttpClient.java [hunk 1]
```diff
@@ -49,14 +49,11 @@ public class ApacheHttpClient implements HttpClient {
...
-    public ApacheHttpClient(Credentials credentials, JsonTransformer jsonTransformer, Map<String, ?> defaultHeaders) {
+    public ApacheHttpClient(Credentials credentials, JsonTransformer jsonTransformer, Map<String, ?> defaultHeaders, CloseableHttpClient httpClient) {
         this.credentials = credentials;
         this.jsonTransformer = jsonTransformer;
         this.defaultHeaders = defaultHeaders;
-        this.httpClient = HttpClientBuilder
-            .create()
-            .setDefaultRequestConfig(RequestConfig.custom().setCookieSpec(CookieSpecs.STANDARD).build())
-            .build();
+        this.httpClient = httpClient;
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/core/http/impl/http/ApacheHttpClient.java:56]: Can we avoid this change? It's not consistent with what we have below in the overloaded version of the constructor.

File: src/main/java/com/crowdin/client/core/http/impl/json/EmptyArrayToNullDeserializer.java [hunk 2]
```diff
@@ -43,11 +42,11 @@ public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
     }
 
     private static boolean isCollectionType(Class<?> type) {
-        return type.isArray() && Collection.class.isAssignableFrom(type);
+        return type.isArray() || Collection.class.isAssignableFrom(type);
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/core/http/impl/json/EmptyArrayToNullDeserializer.java:45]: Please explain this change

File: src/test/java/com/crowdin/client/core/http/CrowdinApiConstructorTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    public CrowdinApiConstructorTest(Credentials credentials, ClientConfig clientConfig) {
+        super(credentials, clientConfig);
+    }
+
+    public CrowdinApiConstructorTest(Credentials credentials) {
+        super(credentials);
+    }
+
+    public HttpClient getHttpClient(){
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/core/http/CrowdinApiConstructorTest.java:17]:
```suggestion
    public HttpClient getHttpClient() {
```

File: src/test/java/com/crowdin/client/core/http/CrowdinApiConstructorTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        return this.httpClient;
+    }
+
+    public String geturl(){
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/core/http/CrowdinApiConstructorTest.java:21]:
```suggestion
    public String geturl() {
```

File: src/test/java/com/crowdin/client/core/http/impl/http/ApacheHttpClientTest.java [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+    private ApacheHttpClient httpClient;
+
+    private final String json = "{\n" +
+            "  \"id\": 123456789,\n" +
+            "  \"title\": \"Example Label\",\n" +
+            "  \"isSystem\": {\n" +
+            "    \"value\": true\n" +
+            "  }\n" +
+            "}\n";
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/core/http/impl/http/ApacheHttpClientTest.java:48]: Typically, we extract such json test data into separate files in the `resources` directory

File: src/main/java/com/crowdin/client/core/http/impl/json/EmptyArrayToNullDeserializer.java (hunk 2 above)
Comment (DecarteAdam) [reply src/main/java/com/crowdin/client/core/http/impl/json/EmptyArrayToNullDeserializer.java:45]: I think these conditions are mutually exclusive in the context of Java:
- If 'type.isArray()' returns' true', it means the given type is an 'array', but arrays in Java are not assignable from 'Collection.class', so 'Collection.class.isAssignableFrom(type') will always return false for 'array' types.
Conversely, if' Collection.class.isAssignableFrom(type)' returns 'true', it means the given type is a subclass of 'Collection', but such types are not arrays, so 'type.isArray()' will always return false.

File: src/main/java/com/crowdin/client/core/http/impl/http/ApacheHttpClient.java (hunk 1 above)
Comment (DecarteAdam) [reply src/main/java/com/crowdin/client/core/http/impl/http/ApacheHttpClient.java:56]: I did that to have the possibilty to mock the 'this.httpClient.request(request)' for my tests. If you have any other possibility how I can do that, I would be glad to hear that :)

File: src/main/java/com/crowdin/client/core/http/impl/json/EmptyArrayToNullDeserializer.java (hunk 2 above)
Comment (andrii-bodnar) [reply src/main/java/com/crowdin/client/core/http/impl/json/EmptyArrayToNullDeserializer.java:45]: Thank you!

File: src/test/java/com/crowdin/client/core/http/impl/json/CrowdinApiExceptionDeserializerTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CrowdinApiExceptionDeserializerTest {
+
+    @Test
+    void testDeserializeHttpBadRequestException() throws IOException {
+        String json = "{\n" +
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/core/http/impl/json/CrowdinApiExceptionDeserializerTest.java:19]: Please also move this to the resources (including the objects below)

**Review by andrii-bodnar — APPROVED** [review]
@DecarteAdam thank you!

**Pull request merged**